src/scope-analyzer.js

```diff
@@ -97,7 +100,8 @@ export default class ScopeAnalyzer extends MonoidalReducer {
   }
 
   reduceClassDeclaration(node, {name, super: _super, elements}) {
-    return super.reduceClassDeclaration(node, {name: name.addDeclarations(DeclarationType.CLASS_DECLARATION), super: _super, elements});
+    let s = super.reduceClassDeclaration(node, {name: name, super: _super, elements}).addDeclarations(DeclarationType.CLASS_NAME).finish(node, ScopeType.CLASS_NAME);
```

Comment:
I'm a little opposed to passing the whole state object on, since doing so would break symmetry with the other `reduce*` methods, most of which need at least one of its properties. Don't know why it ended up not being a shorthand property, though; will correct and force-push.